### DISCUSSION TITLE
In the Console example of Progress how does the API know which Spinner type to initialize?

### DISCUSSION THREAD

**Discussion by svincze** [spectreconsole/spectre.console#841] [Q&A] [answered]
In the Progress example we have the following code: 

```
AnsiConsole.Progress()
            .AutoClear(false)
            .Columns(new ProgressColumn[]
            {
                    new TaskDescriptionColumn(),    // Task description
                    new ProgressBarColumn(),        // Progress bar
                    new PercentageColumn(),         // Percentage
                    new SpinnerColumn(),            // Spinner
            })
            .Start(ctx => { 
            //Omitted...
            }
});
```

How do we know which Spinner type will used? SpinnerColumn initialises one but I see no parameters or possibility to define. Is it always going to be the "Known.Dot" type? 
In short, how do we know what each Column type will define upon object creation? 

Furthermore this is probably my mistake, but I could not find what kind of different column types we have for the "ProgressColumn" object. Can someone show me please?

**Comment by svincze** [comment, marked as answer]
Never mind, I actually found it in the implementation that it will get Spinner.Known.Default which seems it means Dots.

**Reply by MrHinsh** [reply]
Did you figure out how to change it?

**Reply by elgonzo** [reply]
> Did you figure out how to change it?

Take a dive into the API documentation: https://spectreconsole.net/api/spectre.console/spinnercolumn/8147ca48

Spinners included in the Spectre.Console library are available as members of [`Spinner.Known`](https://github.com/spectreconsole/spectre.console/blob/b61fff042bc2e089a9f69709a5171c5f36802921/src/Spectre.Console/Live/Progress/Spinner.Generated.cs#L1568-L1870). A visual preview of many of these spinners can be seen at: https://jsfiddle.net/sindresorhus/2eLtsbey/embedded/result/ (beware, the preview also includes spinners not present in Spinner.Known)
